powerauth-push-server/src/test/java/io/getlime/push/controller/MockPowerAuthController.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2019 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.getlime.push.controller;
+
+import com.google.common.io.BaseEncoding;
+import com.wultra.security.powerauth.client.v3.*;
+import io.getlime.core.rest.model.base.request.ObjectRequest;
+import io.getlime.core.rest.model.base.response.ObjectResponse;
+import io.getlime.push.client.PushServerTestClientFactory;
+import io.getlime.security.powerauth.crypto.lib.generator.KeyGenerator;
+import io.getlime.security.powerauth.crypto.lib.model.exception.CryptoProviderException;
+import io.getlime.security.powerauth.crypto.lib.util.KeyConvertor;
+import org.springframework.context.annotation.Profile;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.security.KeyPair;
+import java.util.*;
+
+/**
+ * Mock controller for PowerAuth services.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@RestController
+@RequestMapping("powerauth-java-server/rest/v3")
+@Profile("test")
+public class MockPowerAuthController {
+
+    private final Set<String> blockedActivations = new HashSet<>();
+
+    @PostMapping("application/list")
+    public ObjectResponse<GetApplicationListResponse> getApplicationList() {
+        final GetApplicationListResponse response = new GetApplicationListResponse();
+        final GetApplicationListResponse.Applications app = new GetApplicationListResponse.Applications();
+        app.setApplicationId(PushServerTestClientFactory.TEST_APPLICATION_NAME);
+        response.getApplications().add(app);
+        return new ObjectResponse<>(response);
+    }
+
+    @PostMapping("application/create")
+    public ObjectResponse<CreateApplicationResponse> createApplication() {
+        final CreateApplicationResponse response = new CreateApplicationResponse();
+        return new ObjectResponse<>(response);
+    }
+
+    @PostMapping("application/detail")
+    public ObjectResponse<GetApplicationDetailResponse> getApplicationDetail() {
```

Comment:
I know, it is only a test, but better not to swallow the exception. But what about
```suggestion
    public ObjectResponse<GetApplicationDetailResponse> getApplicationDetail() throw CryptoProviderException {
```

Reply:
We don't want to declare an exception that can never happen.